osu.Game.Tests/Visual/Multiplayer/TestSceneMultiSpectatorScreen.cs

```diff
@@ -464,16 +463,19 @@ private void waitForCatchup(int userId)
 
         private class TestMultiSpectatorScreen : MultiSpectatorScreen
         {
-            private readonly double? startTime;
+            private readonly double? leadIn;
 
-            public TestMultiSpectatorScreen(Room room, MultiplayerRoomUser[] users, double? startTime = null)
+            public TestMultiSpectatorScreen(Room room, MultiplayerRoomUser[] users, double? leadIn = null)
                 : base(room, users)
             {
-                this.startTime = startTime;
+                this.leadIn = leadIn;
             }
 
             protected override MasterGameplayClockContainer CreateMasterGameplayClockContainer(WorkingBeatmap beatmap)
-                => new MasterGameplayClockContainer(beatmap, 0) { StartTime = startTime ?? 0 };
+            {
+                beatmap.BeatmapInfo.AudioLeadIn = leadIn ?? 0;
```

Comment:
Similarly here - I don't believe setting non-zero lead-in (new code) has anything in common with not starting spectate from time 0 (old code)?

Reply:
Could bring the start time = 0, but think it should rather be irrelevant to rely on it still existing? Though the audio lead-in shouldn't be overwritten to 0 when none is provided...

Reply:
That's missing the point. As stated above the new test looks to be exercising different code paths than the old test, hence me saying that that test coverage of one scenario appears to be getting replaced by coverage of another unrelated scenario, while I would like that think that both should be covered.

Reply:
Well, depends on which old test you're talking about, because if it's the one we have right here on master, it's pretty much invalid to think about. 

Prior to #17302, the "gameplay start time" field would be set as a value to be used in `GameplayClcokContainer.load` directly. Which is what this change is trying to point at, rather than using special properties like `StartTime`.

Reply:
I don't see why the old test would be invalid. It seems perfectly valid to me that you may want to set start time directly rather than going through beatmap lead-in, and both cases should work. In fact `findEarliestStartTime()` [explicitly checks both storyboard events and lead-in time](https://github.com/ppy/osu/blob/26c85161307222319c6254a00a9eb8ac76a0456d/osu.Game/Screens/Play/MasterGameplayClockContainer.cs#L111-L123).

Reply:
Well... it's because setting `StartTime` means updating the gameplay start time prematurely, which means `Reset` calls prior to `MasterGameplayClockContainer.LoadComplete` would seek the clock to the specified start time, which is not something that can happen in real gameplay/spectator/multi-spectator sessions, and was essentially the reason why the test case was false-passing.